### PR TITLE
million checkboxes ui

### PR DESCRIPTION
**Pull request by ldanilek**
customers are requesting an example repo for a component that has a UI, so let's give them one.

shamelessly copied from https://github.com/atrakh/one-million-checkboxes

two small changes:
- it uses the ShardedCounter component to track the count. kind of redundant since the UI already subscribes to all of the data, but if we shifted the query to be paginated this would be useful. and it's a good example of the component.
- create checkbox data lazily instead of requiring you to call `seed`
- 
Screenshot 2024-11-04 at 7 05 59 PM

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by atrakh — APPROVED** [review]

File: example/package.json [hunk 1]
```diff
@@ -4,8 +4,9 @@
   "type": "module",
   "version": "0.0.0",
   "scripts": {
-    "dev": "convex dev --live-component-sources",
+    "dev:backend": "convex dev --live-component-sources",
```
Comment (ianmacartney) [inline example/package.json:7]: shouldn't this have `--typecheck-components` too?

File: example/convex/checkboxes.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+    // Reset the table.
+    for (let i = 0; i < NUM_DOCUMENTS; i++) {
+      await ctx.db.insert("checkboxes", {
+        idx: i,
+        boxes: bytes.buffer,
+      });
+    }
+  },
+});
```
Comment (ianmacartney) [inline example/convex/checkboxes.ts:122]: do you still need this?

Comment (ldanilek) [reply example/convex/checkboxes.ts:122]: nope

File: example/package.json (hunk 1 above)
Comment (ldanilek) [reply example/package.json:7]: is that in the template now? i can add it

**Comment by ianmacartney** [comment]
yes it's in the template. The template has been kept up to date, but
unfortunately it's not trivial to just rebase / merge all the components
(or maybe it is with something like `git patch apply`?)

**Pull request merged**